### PR TITLE
Bugfix + Feature

### PR DESCRIPTION
**Pull request by Moetez16**

### CODE REVIEW

File: grails-app/services/de/laser/SubscriptionService.groovy [hunk 1]
```diff
@@ -1230,10 +1230,10 @@ class SubscriptionService {
                     break
                 case "status": colMap.status = c
                     break
-                case "startdatum":
+                case "laufzeit-beginn":
```
Comment (croqueGrec09) [inline grails-app/services/de/laser/SubscriptionService.groovy:1233]: I would let start/enddatum accepted, too

File: grails-app/services/de/laser/SubscriptionService.groovy [hunk 2]
```diff
@@ -1341,18 +1350,16 @@ class SubscriptionService {
             if(colMap.licenses != null) {
                 List<String> licenseKeys = cols[colMap.licenses].split(',')
                 candidate.licenses = []
-                mappingErrorBag.multipleLicenseError = []
```
Comment (croqueGrec09) [inline grails-app/services/de/laser/SubscriptionService.groovy:1344]: Warning with bulk objects! The error may concern several subscriptions, too!

File: grails-app/services/de/laser/SubscriptionService.groovy [hunk 3]
```diff
@@ -1504,6 +1511,42 @@ class SubscriptionService {
...
             }
+            if(colMap.hasPerpetualAccess != null) {
+                String hasPerpetualAccessKey = cols[colMap.hasPerpetualAccess].trim()
+                if(hasPerpetualAccessKey) {
+                    String yesNo = refdataService.retrieveRefdataValueOID(hasPerpetualAccessKey, RDConstants.Y_N)
+                    if(yesNo) {
+                        candidate.hasPerpetualAccess = yesNo
+                    }
+                    else {
+                        mappingErrorBag.noPerpetualAccessType = hasPerpetualAccessKey
```
Comment (croqueGrec09) [inline grails-app/services/de/laser/SubscriptionService.groovy:1522]: needs to be multiple, too

File: grails-app/services/de/laser/SubscriptionService.groovy [hunk 4]
```diff
@@ -1504,6 +1511,42 @@ class SubscriptionService {
...
+            }
+            if(colMap.hasPublishComponent != null) {
+                String hasPublishComponentKey = cols[colMap.hasPublishComponent].trim()
+                if(hasPublishComponentKey) {
+                    String yesNo = refdataService.retrieveRefdataValueOID(hasPublishComponentKey, RDConstants.Y_N)
+                    if(yesNo) {
+                        candidate.hasPublishComponent = yesNo
+                    }
+                    else {
+                        mappingErrorBag.noPublishComponent = hasPublishComponentKey
```
Comment (croqueGrec09) [inline grails-app/services/de/laser/SubscriptionService.groovy:1534]: needs to be multiple, too

File: grails-app/services/de/laser/SubscriptionService.groovy [hunk 5]
```diff
@@ -1504,6 +1511,42 @@ class SubscriptionService {
...
+            }
+            if(colMap.isPublicForApi != null) {
+                String isPublicForApiKey = cols[colMap.isPublicForApi].trim()
+                if(isPublicForApiKey) {
+                    String yesNo = refdataService.retrieveRefdataValueOID(isPublicForApiKey, RDConstants.Y_N)
+                    if(yesNo) {
+                        candidate.isPublicForApi = yesNo
+                    }
+                    else {
+                        mappingErrorBag.noPublicForApi = isPublicForApiKey
```
Comment (croqueGrec09) [inline grails-app/services/de/laser/SubscriptionService.groovy:1546]: needs to be multiple, too

File: grails-app/services/de/laser/SubscriptionService.groovy [hunk 6]
```diff
@@ -1524,7 +1567,7 @@ class SubscriptionService {
                 candidate.properties[k] = propData
             }
             //notes
-            if(colMap.notes != null && cols[colMap.notes].trim()) {
+            if(colMap.notes != null && cols[colMap.notes]) {
```
Comment (croqueGrec09) [inline grails-app/services/de/laser/SubscriptionService.groovy:1570]: You permit empty values

**Review by croqueGrec09 — CHANGES_REQUESTED** [review]
cases when errors concern multiple subscriptions are not considered

File: grails-app/services/de/laser/SubscriptionService.groovy [hunk 7]
```diff
@@ -1341,7 +1352,7 @@ class SubscriptionService {
             if(colMap.licenses != null) {
                 List<String> licenseKeys = cols[colMap.licenses].split(',')
                 candidate.licenses = []
-                mappingErrorBag.multipleLicenseError = []
+                mappingErrorBag.multipleLicError = []
```
Comment (croqueGrec09) [inline grails-app/services/de/laser/SubscriptionService.groovy:1355]: Is output corrected?

Comment (Moetez16) [reply grails-app/services/de/laser/SubscriptionService.groovy:1355]: Yes, the message property has this name.

**Comment by Moetez16** [comment]
PR geupdatet with new Bugfixes

**Review by croqueGrec09 — APPROVED** [review]
approved

**Pull request merged**